crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -97,7 +100,7 @@ impl Solution {
 
     /// The ID of this solution.
     pub fn id(&self) -> Id {
-        self.id
+        self.id.clone()
```

Comment:
nit: We should probably return a reference and let the caller decide if they actually need a copy.